dart/dynamics/Entity.h

```diff
@@ -189,6 +198,19 @@ class Entity : public virtual common::Subject
   bool needsAccelerationUpdate() const;
 
 protected:
+
+  /// Used when constructing a Frame class, because the Frame constructor will
+  /// take care of setting up the parameters you pass into it
+  enum ConstructFrame_t { ConstructFrame };
+
+  explicit Entity(ConstructFrame_t);
```

Comment:
In this specific case, it definitely seems necessary, to distinguish the name of the enum type from the name of the enum itself. I think in general I *tend* to end enum type names with a ``_t`` suffix, but only because that's a conventional thing to do; not because I feel strongly about it.